packages/template-retail-react-app/app/hooks/use-pickup-shipment.js

```diff
@@ -169,9 +185,72 @@ export const usePickupShipment = () => {
         })
     }
 
+    /**
+     * Configure shipping method based on pickup selection
+     * @param {Object} basketResponse - The basket response from adding items
+     * @param {Array} productItems - Array of product items that were added
+     * @param {boolean} hasAnyPickupSelected - Whether any items have pickup selected
+     * @param {Object} selectedStore - The selected store information
+     * @returns {Promise<void>}
+     */
+    const configureShippingMethodIfNeeded = async (
```

Comment:
Moved this helper into the `usePickupShipment` hook so we can keep pickup login outside of the `checkout-context` file